packages/ui/VirtualList/VirtualListBase.js

```diff
@@ -391,16 +393,28 @@ const VirtualListBaseFactory = (type) => {
 					x: xMax > x ? x : xMax,
 					y: yMax > y ? y : yMax
 				});
+
+				this.deferScrollTo = true;
 			} else if (this.hasDataSizeChanged) {
 				const newState = this.getStatesAndUpdateBounds(this.props, this.state.firstIndex);
 				// eslint-disable-next-line react/no-did-update-set-state
 				this.setState(newState);
 				this.setContainerSize();
+
+				this.deferScrollTo = true;
 			} else if (prevProps.rtl !== this.props.rtl) {
 				this.updateScrollPosition(this.getXY(this.scrollPosition, 0));
 			}
+
+			const maxPos = this.isPrimaryDirectionVertical ? this.scrollBounds.maxTop : this.scrollBounds.maxLeft;
+
+			if (!this.deferScrollTo && this.scrollPosition > maxPos) {
+				this.props.cbScrollTo({position: (this.isPrimaryDirectionVertical) ? {y: maxPos} : {x: maxPos}, animate: false});
+			}
 		}
 
+		deferScrollTo = false
```

Comment:
I don't _think_ we need this to be a class member since it's set and only read within `componentDidUpdate`. Am I reading that correctly?

Reply:
Yes. I'll fix it. Thanks.

Reply:
You need to fix your clock, it's too early to be up!